src/sagemaker/jumpstart/curated_hub/curated_hub.py

```diff
@@ -113,3 +173,241 @@ def delete_model(self, model_name: str, model_version: str = "*") -> None:
+        """Returns list of `HubContent` that have been created from a JumpStart model."""
+        hub_models = self.list_models()
+
+        js_models_in_hub = []
+        for hub_model in hub_models["HubContentSummaries"]:
+            # TODO: extract both in one pass
+            jumpstart_model_id = next(
+                (
+                    tag
+                    for tag in hub_model["search_keywords"]
+                    if tag.startswith(JUMPSTART_HUB_MODEL_ID_TAG_PREFIX)
+                ),
+                None,
+            )
+            jumpstart_model_version = next(
+                (
+                    tag
+                    for tag in hub_model["search_keywords"]
+                    if tag.startswith(JUMPSTART_HUB_MODEL_VERSION_TAG_PREFIX)
+                ),
+                None,
+            )
+
+            if jumpstart_model_id and jumpstart_model_version:
+                js_models_in_hub.append(hub_model)
+
+        return js_models_in_hub
+
+    def _determine_models_to_sync(
+        self, model_list: List[JumpStartModelInfo], models_in_hub
+    ) -> List[JumpStartModelInfo]:
+        """Determines which models from `sync` params to sync into the CuratedHub.
+
+        Algorithm:
+
+            First, look for a match of model name in Hub. If no model is found, sync that model.
+
+            Next, compare versions of model to sync and what's in the Hub. If version already
+            in Hub, don't sync. If newer version in Hub, don't sync. If older version in Hub,
+            sync that model.
+        """
+        models_to_sync = []
+        for model in model_list:
+            matched_model = next(
+                (
+                    hub_model
+                    for hub_model in models_in_hub
+                    if hub_model and hub_model["name"] == model.model_id
+                ),
+                None,
+            )
+
+            # Model does not exist in Hub, sync
+            if not matched_model:
+                models_to_sync.append(model)
+
+            if matched_model:
+                model_version = Version(model.version)
+                hub_model_version = Version(matched_model["version"])
+
+                # 1. Model version exists in Hub, pass
+                if hub_model_version == model_version:
+                    pass
+
+                # 2. Invalid model version exists in Hub, pass
+                # This will only happen if something goes wrong in our metadata
+                if hub_model_version > model_version:
+                    pass
+
+                # 3. Old model version exists in Hub, update
+                if hub_model_version < model_version:
+                    # Check minSDKVersion against current SDK version, emit log
+                    models_to_sync.append(model)
+
+        return models_to_sync
+
+    def sync(self, model_list: List[Dict[str, str]]):
+        """Syncs a list of JumpStart model ids and versions with a CuratedHub
```

Comment:
This is the main customer-facing utility, we should improve the docstring. How about:

```py

"""Find existing models in the Hub, compare with requested list, add or remove entries.

First, call ListHubContents on the Hub, then determine the change set required. 
Then for each model in the change set: 
- call ImportHubContent to add a new model to the hub if missing
- call ImportHubContent to add a new version if the model is present in the hub, but not that version,
- call DeleteHubContent for all existing versions in the Hub if the model was removed from the requested list.

Args:
  ...
Raises:
  ...
"""
```